frontend/src/layout/Header.tsx

```diff
@@ -24,39 +26,111 @@ export const Header: React.FC = () => (
-            "& > *": { height: "100%", padding: "4px 0" },
-        }}>
+        <picture css={{ height: "100%" }}>
             <source media="(min-width: 450px)" srcSet="/assets/static/logo-large.svg" />
-            <img src="/assets/static/logo-small.svg" />
+            <img
+                css={{ height: "100%", padding: "4px 8px" }}
+                src="/assets/static/logo-small.svg"
+            />
         </picture>
     </Link>
 );
 
 
 const SEARCH_HEIGHT = 35;
 
-const Search: React.FC = () => (
-    <input
-        type="text"
-        placeholder="Search"
-        css={{
-            flex: "1 1 0px",
-            margin: "0 16px",
-            minWidth: 50,
-            maxWidth: 280,
-            height: SEARCH_HEIGHT,
-            borderRadius: SEARCH_HEIGHT / 2,
-            border: "1.5px solid #ccc",
-            padding: `0 ${SEARCH_HEIGHT / 2}px`,
-        }}
-    />
-);
+const Search: React.FC = () => {
+    const { t } = useTranslation();
 
-const Menu: React.FC = () => (
-    <div>
-        Not logged in
-        <FontAwesomeIcon css={{ marginLeft: 4 }} icon={faCaretDown} />
-    </div>
-);
+    return (
+        <input
+            type="text"
+            placeholder={t("search")}
+            css={{
+                flex: "1 1 0px",
+                margin: "0 8px",
+                minWidth: 50,
+                maxWidth: 280,
+                height: SEARCH_HEIGHT,
+                borderRadius: SEARCH_HEIGHT / 2,
+                border: "1.5px solid #ccc",
+                padding: `0 ${SEARCH_HEIGHT / 2}px`,
+            }}
+        />
+    );
+};
+
+const Menu: React.FC = () => {
+    const { t, i18n } = useTranslation();
+
+    type MenuState = "closed" | "language";
+    const [menuState, setMenuState] = useState<MenuState>("closed");
+    const toggleMenu = (state: MenuState) => {
+        setMenuState(menuState === state ? "closed" : state);
+    };
+
+    const menuContent = (() => {
+        switch (menuState) {
+            case "closed": return null;
+            case "language": return (
+                <ul css={{
+                    width: "100%",
+                    listStyle: "none",
+                    margin: 0,
+                    padding: 0,
+                }}>
+                    {Object.keys(languages).map(lng => (
+                        <li
+                            onClick={() => i18n.changeLanguage(lng)}
+                            key={lng}
+                            css={{
+                                padding: "4px 8px 4px 16px",
+                                cursor: "pointer",
+                                "&:hover": {
+                                    backgroundColor: "#ddd",
+                                },
+                            }}
+                        >{t("language-name", { lng })}</li>
+                    ))}
+                </ul>
+            );
+            // TODO: extract this helper function and give it a good name.
```

Comment:
![](https://i.pinimg.com/originals/63/f6/a2/63f6a2e4de1e1bb219b1ec51072fdbc6.gif)

Reply:
In this PR? :D

Reply:
Why not? Let's say "yes". :stuck_out_tongue_closed_eyes: